ddtrace/internal/packages.py

```diff
@@ -46,15 +46,19 @@ def get_distributions() -> t.Mapping[str, str]:
 def get_package_distributions() -> t.Mapping[str, t.List[str]]:
     """a mapping of importable package names to their distribution name(s)"""
     global _PACKAGE_DISTRIBUTIONS
-    if _PACKAGE_DISTRIBUTIONS is None:
+    if not _PACKAGE_DISTRIBUTIONS:
         try:
             import importlib.metadata as importlib_metadata
         except ImportError:
             import importlib_metadata  # type: ignore[no-redef]
 
         # Prefer the official API if available, otherwise fallback to the vendored version
         if hasattr(importlib_metadata, "packages_distributions"):
-            _PACKAGE_DISTRIBUTIONS = importlib_metadata.packages_distributions()
+            try:
+                _PACKAGE_DISTRIBUTIONS = importlib_metadata.packages_distributions()
+            except Exception:
```

Comment:
with this, you could probably just drop the `hasattr`, but 🤷🏻

Reply:
It's related to the issue #APMAPI-1400 where in some circonstances on system tests we get an OSError.